### DISCUSSION TITLE
selectionDisabled on Datatable sets selection to null after commandButton press - Primefaces 14.0.0

### DISCUSSION THREAD

**Discussion by hangrybear666** [primefaces/community#2198] [PrimeFaces] [answered]
When p:dataTable selection is disabled via `selectionDisabled="true"` and trying to save user state via a `p:commandButton` `actionListener` the selection object in my backing bean is nullified and saving to the database fails.

This is a **regression** from our prior version Primefaces 10.0 where the `disabledSelection` flag (which got renamed to `selectionDisabled` now..) did **not** nullify the selection object in my backing bean.

Do I have to migrate away from using the `selectionDisabled` flag, or is this unintended behavior?

``` 
<p:dataTable id="dtEffectiveDate"
        class="dt-mark-selection dt-slim-vertical dt-no-column-header mb-2"
        binding="#{federalStateWorkTimeModel.effectiveDatesDataTable}"
        selection="#{federalStateWorkTimeModel.effectiveDate}"
        rowKey="#{item.id}"
        selectionDisabled="#{federalStateWorkTimeModel.inEditMode}"
        showGridlines="false" var="item"
        value="#{federalStateWorkTimeModel.effectiveDates}"
        widgetVar="dtEffectiveDates">


<p:commandButton class="generic-btn generic-btn-yes mt-2"
      value="#{commonRes.common_save}" icon="pi pi-save"
      immediate="false" tabindex="-1"
      rendered="#{federalStateWorkTimeModel.inEditMode}"
      style="width:100%;"
      actionListener="#{federalStateWorkTimeModel.saveChanges}"
      update="idForm" 
/>
``` 

**Edit: Workaround solution is to disable selection client side on demand via e.g. oncomplete callbacks**

.xhtml
```
<p:commandButton 
   [...]
   oncomplete="disableRowSelection('dataTableId')"
/>
      
/>
```

.js
```
function disableRowSelection(dataTableId) {
    const rows = Array.from(
  		document.querySelectorAll('[id*="' + dataTableId + '"].ui-datatable')[0]
  			.getElementsByTagName('tr')
 		);
    rows.forEach(row => {
        const handler = function(event) {
            event.stopPropagation();
        };
    	row.addEventListener("click", handler);
    })
}
```

**Comment by melloware** [comment]
cc @Rapster 

I am pretty sure this was intentional with this security ticket: https://github.com/primefaces/primefaces/issues/6682

There was also this discussion: https://github.com/primefaces/primefaces/issues/11356

Please provide an executable example using the [PrimeFaces Test](https://github.com/primefaces/primefaces-test) project.  It is the only way developers can debug your problem to help.

**Reply by hangrybear666** [reply]
Thank you for the quick reply. Should I still create a test project next week when I'm back to work or is this intended as described in your links and I have to find a workaround?

My disabled and selected value is supposed to be persisted in db, hence the question.

**Reply by melloware** [reply]
Yes I want to see if what you are doing is the same issue. Or something else different.

**Reply by hangrybear666** [reply]
Will do next week and update the thread

**Comment by hangrybear666** [comment]
@melloware  I've created a primefaces-test project that is self-explanatory.

**MP4 Screen Capture displaying workflow:**
https://github.com/primefaces/community/assets/85507965/b2ebb023-565f-4c8f-a190-55f049fab61d

**Reproducer Primfaces v14 Test Archive**
[primefaces-test-selectionDisabled-nullifies.zip](https://github.com/user-attachments/files/16072737/primefaces-test-selectionDisabled-nullifies.zip)

**Reply by melloware** [reply, marked as answer]
OK from watching your video i think this is the issue: https://github.com/primefaces/primefaces/issues/6682  When selectionDisabled="true" which its while you are editing when you click Cancel or Save its submitting the Form to the backing bean and the DT sees the selectionisDisabled and does not allow Selection values per the ticket #6882 which would allow an attacker to just form POST the values if we don't respect that flag.

Another option you have is just to disable selection with client side Javascript and re-enable it when you press cancel or save.

**Reply by Rapster** [reply]
Maybe at some point, we'll have to come up with a solution, because it's been reported more than once... Something like `selectionExcluded` which does exactly what `selectionDisabled` already do, and `selectionDisabled` would be a client hack (solution you just propose)

**Reply by hangrybear666** [reply]
Thank you for the quick response.

I have added eventListener to tr Elements within the datatable, listening for a click and stopping propagation, hence disabled selection client side. This can be toggled on demand via e.g. commandbutton oncomplete callback.

I will edit this as a workaround to my initial post and this topic can be closed
